package maidsafe.vault.protobuf;
option optimize_for = LITE_RUNTIME;

message PmidRecord {
  required bytes pmid_name = 1;
  required int64 stored_count = 2;
  required int64 stored_total_size = 3;
  required int64 lost_count = 4;
  required int64 lost_total_size = 5;
  required int64 claimed_available_size = 6;
}

message PmidAccountDbValue {
  required int32 size = 1;
}

message PmidAccountDbEntry {
  required int32 type = 1;  // name and type == db key
  required bytes name = 2;
  required PmidAccountDbValue value = 3;
}

message PmidAccount {
  required bytes serialised_account_details = 1;
}

message PmidAccountDetails {
  required bytes serialised_pmid_record = 1;
  repeated PmidAccountDbEntry db_entry = 2;
  repeated bytes serialised_unresolved_entry = 3;
}


